File: FPTalk1/Startup.cs
using System;
using System.IO;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.DependencyInjection;
using DiscriminatedUnion;
using static FPTalk1.Content;
using static DiscriminatedUnion.MatchExtensions;

//using contentType = DiscriminatedUnion
//	.Union<FPTalk1.Content.Json,
//		   FPTalk1.Content.Url,
//		   FPTalk1.Content.Plain,
//		   FPTalk1.Content.OctetStream>;
using static FPTalk1.RequestMethod;

namespace FPTalk1
{
	public class Startup
	{

		readonly Func<HttpContext, Func<Task>, Task> UnitHandler = (context, next) => next();
		
		// This method gets called by the runtime. Use this method to add services to the container.
		// For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
		public void ConfigureServices(IServiceCollection services)
		{
		}

		// This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
		public void Configure(IApplicationBuilder app, IHostingEnvironment env)
		{
			if (env.IsDevelopment())
			{
				app.UseDeveloperExceptionPage();
			}

			var filter = MethodFilter(RequestMethod.POST).And(PathFilter("/"));
			
			app.Use(Middleware(filter, fun((string a) => $"Hello {a}").Jsonify()));
			app.Use(Middleware(h => true, fun((string a) => $"Goodbye").Jsonify()));
		}

	}
}

File: FPTalk2/Startup.fs
namespace FPTalk2


open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Hosting
open Microsoft.Extensions.Configuration
open Microsoft.Extensions.DependencyInjection
open Microsoft.AspNetCore.Http
open Content

//type Middleware = (HttpContext-> bool) -> (HttpContext -> Task) -> HttpContext -> Func<Task> -> Task
type Startup private () =
    let ( <&> ) a b = Middleware a b
    let ( <+> ) (a:Filter) (b:Filter) = fun h -> (a h) && (b h)

    new (configuration: IConfiguration) as this =
        Startup() then
        this.Configuration <- configuration


    // This method gets called by the runtime. Use this method to add services to the container.
    member this.ConfigureServices(services: IServiceCollection) =
        // Add framework services.
        services.AddMvc() |> ignore

    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
    member this.Configure(app: IApplicationBuilder, env: IHostingEnvironment) =
        let register = useEx app
        
        register ((MethodFilter RequestMethod.Post <+> PathFilter "/") <&> (Jsonify (fun c -> "Hello " + c)))
        
        register ((fun c -> true) <&> (Jsonify (fun c -> "Goodbye" + c)))

        app.UseMvc() |> ignore

    member val Configuration : IConfiguration = null with get, set